device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -952,41 +952,40 @@ private void checkForNewlyUnregisteredMultiplexedClientsToStop()
-            }
-            else
-            {
-                log.warn("Attempted to remove device session for device {} from multiplexed connection, but device was not currently registered.", configToUnregister.getDeviceId());
-                return;
-            }
 
-            // Need to find the sas token renewal handler that is tied to this device
-            AmqpsSasTokenRenewalHandler sasTokenRenewalHandlerToRemove = null;
-            for (AmqpsSasTokenRenewalHandler existingSasTokenRenewalHandler : this.sasTokenRenewalHandlers)
-            {
-                if (existingSasTokenRenewalHandler.amqpsSessionHandler.getDeviceId().equals(configToUnregister.getDeviceId()))
+                // Need to find the sas token renewal handler that is tied to this device
+                AmqpsSasTokenRenewalHandler sasTokenRenewalHandlerToRemove = null;
+                for (AmqpsSasTokenRenewalHandler existingSasTokenRenewalHandler : this.sasTokenRenewalHandlers)
                 {
-                    sasTokenRenewalHandlerToRemove = existingSasTokenRenewalHandler;
+                    if (existingSasTokenRenewalHandler.amqpsSessionHandler.getDeviceId().equals(configToUnregister.getDeviceId()))
+                    {
+                        sasTokenRenewalHandlerToRemove = existingSasTokenRenewalHandler;
 
-                    // Stop the sas token renewal handler from sending any more authentication messages on behalf of this device
-                    log.trace("Closing sas token renewal handler for device {}", configToUnregister.getDeviceId());
-                    sasTokenRenewalHandlerToRemove.close();
-                    break;
+                        // Stop the sas token renewal handler from sending any more authentication messages on behalf of this device
+                        log.trace("Closing sas token renewal handler for device {}", configToUnregister.getDeviceId());
+                        sasTokenRenewalHandlerToRemove.close();
+                        break;
+                    }
                 }
-            }
 
-            if (sasTokenRenewalHandlerToRemove != null)
-            {
-                this.sasTokenRenewalHandlers.remove(sasTokenRenewalHandlerToRemove);
-            }
+                if (sasTokenRenewalHandlerToRemove != null)
+                {
+                    this.sasTokenRenewalHandlers.remove(sasTokenRenewalHandlerToRemove);
+                }
 
-            this.reconnectionsScheduled.remove(configToUnregister.getDeviceId());
+                this.reconnectionsScheduled.remove(configToUnregister.getDeviceId());
 
-            log.debug("Closing device session for multiplexed device {}", configToUnregister.getDeviceId());
-            amqpsSessionHandler.closeSession();
+                log.debug("Closing device session for multiplexed device {}", configToUnregister.getDeviceId());
+                amqpsSessionHandler.closeSession();
+            }
+            else
```

Comment:
consider handling the error/smaller case first, so both blocks appear closer to the if condition. Otherwise, by the time a reader gets down here they've lost the context of what `else` means.